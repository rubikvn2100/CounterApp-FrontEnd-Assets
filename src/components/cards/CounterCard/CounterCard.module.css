.card {
  width: 320px;
  display: flex;
  flex-direction: column;
  border: 1px solid #0000007f;
  border-radius: 1.5rem;
  background-color: #f4ece6f0;
  box-shadow: -10px 10px 10px #0000003f;
  padding: 1.5rem;
}

.counterDisplay {
  font-size: 2rem;
}

.incrementButton {
  margin-top: 2rem;
  border: 2px solid #ffffff;
  border-radius: 1000px;
  background-color: #ffffff;
  padding: 0.75rem 1.5rem;
  text-align: center;
  font-size: 0.875rem;
  color: #000000;
  transition: all 0.2s ease;
}

@media (hover: hover) {
  .incrementButton:hover {
    color: #777777;
    background-color: transparent;
  }
}

.incrementButton:active {
  color: #777777;
  background-color: transparent;
}
