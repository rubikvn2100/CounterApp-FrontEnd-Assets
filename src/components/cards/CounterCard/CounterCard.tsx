import React, { useState } from "react";
import styles from "./CounterCard.module.css";

const CounterCard: React.FC = () => {
  const [count, setCount] = useState(0);

  const increment = () => {
    setCount(count + 1);
  };

  return (
    <div className={styles.card}>
      <div className={styles.counterDisplay}>Counter: {count}</div>
      <div onClick={increment} className={styles.incrementButton}>
        Increment
      </div>
    </div>
  );
};

export default CounterCard;
