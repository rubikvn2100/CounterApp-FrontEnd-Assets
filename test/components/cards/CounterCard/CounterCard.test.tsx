import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import CounterCard from "../../../../src/components/cards/CounterCard/CounterCard";

describe("CounterCard", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<CounterCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("increments counter on button click", () => {
    const { getByText } = render(<CounterCard />);
    const incrementButton = getByText(/Increment/i);
    const counterDisplay = getByText(/Counter: 0/i);

    fireEvent.click(incrementButton);

    expect(counterDisplay).toHaveTextContent("Counter: 1");
  });
});
