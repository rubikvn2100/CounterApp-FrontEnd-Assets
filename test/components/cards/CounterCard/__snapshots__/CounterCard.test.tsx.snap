// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterCard renders correctly 1`] = `
<div>
  <div>
    Counter: 
    0
  </div>
  <div
    onClick={[Function]}
  >
    Increment
  </div>
</div>
`;
